src/main/java/gregtech/api/metatileentity/MetaTileEntity.java

```diff
@@ -1360,7 +1361,16 @@ public RecipeMap<?> getRecipeMap() {
     public void doExplosion(float explosionPower) {
         getWorld().setBlockToAir(getPos());
         getWorld().createExplosion(null, getPos().getX() + 0.5, getPos().getY() + 0.5, getPos().getZ() + 0.5,
-                explosionPower, ConfigHolder.machines.doExplosions);
+                explosionPower, ConfigHolder.machines.doExplosionDamagesTerrain);
+    }
+
+    public void setOnFire() {
+        for (EnumFacing side : EnumFacing.VALUES) {
+            if (getWorld().isAirBlock(getPos().offset(side))) {
+                getWorld().setBlockState(getPos().offset(side), Blocks.FIRE.getDefaultState(), 11);
+                return;
```

Comment:
Instead of calling `return` here, which will result in only 1 air space being turned to fire, this method could take in a double for percentage chance to turn blocks to fire. It could look something like

```java
public void setOnFire(double chance) {
        for(EnumFacing side : EnumFacing.VALUES) {
        if(chance >= GTValues.RNG.nextDouble(100.00) && getWorld.isAirBlock(getPos().offset(side))) {
        getWorld().setBlockState(getPos().offset(side), Blocks.FIRE.getDefaultState(), 11);
        }
    }
}
```

This would allow for higher tier machines a chance for setting more sides on fire at one time, if the passed chance is based on the tier of the machine.

However, this does not guarantee at least one fire spawned, so you could also change the above method and have the passed `chance` value be the chance for an additional fire to be spawned after the first fire is spawned. (I think this makes the most sense)